rts/lra/lra-coordinator-jar/src/test/java/io/narayana/lra/coordinator/TestBase.java

```diff
@@ -0,0 +1,207 @@
+    private static String recoveryUrl;
+    private static String jvmArgs;
+    private static Path storeDir;
+    private static String deploymentURL;
+
+    NarayanaLRAClient lraClient;
+
+    @ArquillianResource
+    private ContainerController containerController;
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    @BeforeClass
+    public static void beforeClass() {
+        String host = System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_HOST_KEY, "localhost");
+        String port = System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_PORT_KEY, "8080");
+        String path = System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_PATH_KEY, COORDINATOR_PATH_NAME);
+
+        coordinatorUrl = String.format("http://%s:%s/%s", host, port, path);
+        recoveryUrl = String.format("http://%s:%s/%s/%s/%s",
+                host, port, COORDINATOR_DEPLOYMENT, RECOVERY_COORDINATOR_PATH_NAME, RECOVERY_COORDINATOR_SUB_RESOURCE_NAME);
+        storeDir = Paths.get(String.format("%s/standalone/data/tx-object-store", System.getProperty("env.JBOSS_HOME", "null")));
+        jvmArgs = System.getProperty("server.jvm.args");
+
+        deploymentURL = String.format("http://%s:%s/%s", host, port, COORDINATOR_DEPLOYMENT);
+    }
+
+    @Before
+    public void before() throws URISyntaxException, MalformedURLException {
+        LRALogger.logger.infof("Starting test %s", testName);
+
+        lraClient = new NarayanaLRAClient(new URI(coordinatorUrl));
+    }
+
+    @After
+    public void after() {
+        LRALogger.logger.infof("Finised test %s", testName);
+        lraClient.close();
+    }
+
+    void startContainer(String bytemanScript) {
+        Config config = new Config();
+        String javaVmArguments = System.getProperty("server.jvm.args");
+
+        clearRecoveryLog();
+
+        if (bytemanScript != null) {
+            javaVmArguments = javaVmArguments.replaceAll("=listen", "=script:scripts/@BMScript@.btm,listen");
+            javaVmArguments = javaVmArguments.replace("@BMScript@", bytemanScript);
+        }
+
+        config.add("javaVmArguments", javaVmArguments);
+
+        containerController.start(COORDINATOR_CONTAINER, config.map());
+        deployer.deploy(COORDINATOR_DEPLOYMENT);
+    }
+
+    void restartContainer() {
+        try {
+            // ensure that the controller is not running
+            containerController.kill(COORDINATOR_CONTAINER);
+            System.out.printf("jboss-as kill worked%n");
+        } catch (Exception e) {
+            System.out.printf("jboss-as kill: %s", e.getMessage());
+        }
+
+        Config config = new Config();
+        String javaVmArguments = System.getProperty("server.jvm.args");
+        config.add("javaVmArguments", javaVmArguments);
+        containerController.start(COORDINATOR_CONTAINER);
+    }
+
+    void stopContainer() {
+        if (containerController.isStarted(COORDINATOR_CONTAINER)) {
+            LRALogger.logger.info("STOPPING CONTAINER%n");
```

Comment:
infof?

Reply:
I was going to mention that printf are probably OK in the tests. But I think it is a bit confusing that there is a mix of printf and the logger. I think that using the LRA internal logger could be confusing in the output

Reply:
+1 I removed all system printf's and fixed this format string on the logger